### PR TITLE
[Shop] Sort product by date 

### PR DESCRIPTION
**Pull request by tuka217**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Related tickets | -
| License         | MIT

### CODE REVIEW

File: features/product/viewing_products/viewing_product_sorted_by_date.feature [hunk 1]
```diff
@@ -0,0 +1,27 @@
+@viewing_products
+Feature: Sorting listed products
+    In order to change the order by which products are displayed
+    As an Administrator
+    I want to sort products
```
Comment (Zales0123) [inline features/product/viewing_products/viewing_product_sorted_by_date.feature]: ``I want to be able to sort products``?

File: features/product/viewing_products/viewing_product_sorted_by_date.feature [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        And this product was created at "05-10-2016"
+        And this product belongs to "Fluffy Pets"
+        And the store also has a product "Pug of Love" with code "L_PUG"
+        And this product was created at "06-10-2016"
+        And this product belongs to "Fluffy Pets"
+        And the store also has a product "Xtreme Pug" with code "X_PUG"
+        And this product was created at "07-09-2016"
+        And this product belongs to "Fluffy Pets"
+        And I am a logged in customer
+
```
Comment (Zales0123) [inline features/product/viewing_products/viewing_product_sorted_by_date.feature]: Redundant blank line.

File: features/product/viewing_products/viewing_product_sorted_by_date.feature [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+    @todo
+    Scenario: Sorting products by their dates of creations
+        When I browse products from taxon "Fluffy Pets"
+        And I start sorting products by creation date
```
Comment (Zales0123) [inline features/product/viewing_products/viewing_product_sorted_by_date.feature]: ``And I sort products by theirs creation dates``? ``start`` seems like it's a long-time process 😄

File: src/Sylius/Behat/Context/Setup/ProductContext.php [hunk 4]
```diff
@@ -664,4 +664,14 @@ private function getParameter($name)
     {
         return isset($this->minkParameters[$name]) ? $this->minkParameters[$name] : null;
     }
+
+    /**
+     * @Given this product was created at :date
+     */
+    public function thisProductWasCreatedAt($date)
+    {
+        $dateOfCreation = date_create_from_format('d-m-Y', $date);
```
Comment (Zales0123) [inline src/Sylius/Behat/Context/Setup/ProductContext.php]: Why not just ``$dateOfCreation = new \DateTime($data);``?

File: features/product/viewing_products/viewing_product_sorted_by_date.feature [hunk 5]
```diff
@@ -0,0 +1,27 @@
+@viewing_products
+Feature: Sorting listed products
+    In order to change the order by which products are displayed
+    As an Administrator
+    I want to sort products
+
+    Background:
+        Given the store operates on a single channel in "United States"
+        And the store has a product "Berserk Pug" with code "B_PUG"
+        And this product was created at "05-10-2016"
```
Comment (Zales0123) [inline features/product/viewing_products/viewing_product_sorted_by_date.feature]: Maybe we could merge this two steps into ``And the store has a product "Berserk Pug" with code "B_PUG", created at "05-10-2016"``? I would significantly reduce background size.

File: features/product/viewing_products/viewing_product_sorted_by_date.feature (hunk 3 above)
Comment (tuka217) [inline features/product/viewing_products/viewing_product_sorted_by_date.feature]: ☁️  ⛰  🐴

File: features/product/viewing_products/viewing_product_sorted_by_date.feature [hunk 6]
```diff
@@ -0,0 +1,23 @@
+@viewing_products
+Feature: Sorting listed products
+    In order to change the order by which products are displayed
```
Comment (TheMadeleine) [inline features/product/viewing_products/viewing_product_sorted_by_date.feature]: **in** which

**Comment by pjedrzejewski** [comment]
Thank you Ania! :) 👍

**Pull request merged**